### PR TITLE
The 304 status code processing (close #817, close #617)

### LINKED ISSUE #617: "Uncaught ReferenceError: $ is not defined" on mercadolibre.com.ar
This error occurs on every second page load only in Crome.

### LINKED ISSUE #817: Site http://www.avg.com/ru-ru/homepage is hungs
see [vide](http://screencast.com/t/rUXG14NbK145)
It looks like a problem with 3хх status codes.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev @VasilyStrelyaev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
Wait for #827

**Comment by churkin** [comment]
@testcafe-build-bot \retest

File: src/request-pipeline/context.js [hunk 1]
```diff
@@ -227,6 +227,11 @@ export default class RequestPipelineContext {
         this.res.end();
     }
 
+    isNotModified () {
+        return this.req.method === 'GET' && this.destRes.statusCode === 304 &&
```
Comment (miherlosev) [inline src/request-pipeline/context.js]: It is used in two places. So we can remake this code as property `RequestPipelineContext.isNotModified`.

**Comment by LavrovArtem** [comment]
FPR

File: src/request-pipeline/context.js [hunk 2]
```diff
@@ -227,6 +227,11 @@ export default class RequestPipelineContext {
         this.res.end();
     }
 
+    checkNotModifiedStatus () {
+        this.isNotModified = this.req.method === 'GET' && this.destRes.statusCode === 304 &&
```
Comment (churkin) [inline src/request-pipeline/context.js]: I think we should move it to the `buildContentInfo` method

File: src/request-pipeline/context.js [hunk 3]
```diff
@@ -172,8 +172,10 @@ export default class RequestPipelineContext {
         var isRedirect              = this.destRes.headers['location'] &&
                                       REDIRECT_STATUS_CODES.indexOf(this.destRes.statusCode) > -1;
         var requireAssetsProcessing = (isCSS || isScript || isManifest) && this.destRes.statusCode !== 204;
+        var isNotModified           = this.req.method === 'GET' && this.destRes.statusCode === 304 &&
+                                      (this.req.headers['if-modified-since'] || this.req.headers['if-none-match']);
```
Comment (miherlosev) [inline src/request-pipeline/context.js]: @LavrovArtem @churkin 
For consistency, property definition should be moved to constructor and property calcualation to the `_initRequestNatureInfo`.

Comment (LavrovArtem) [inline src/request-pipeline/context.js]: I'm not sure. Context has no the destRes object into the _initRequestNature info the function.

Comment (miherlosev) [inline src/request-pipeline/context.js]: Agree with you.

**Comment by LavrovArtem** [comment]
FPR

**Pull request merged**